resource-service/src/main/java/uk/gov/gchq/palisade/service/resource/model/AuditableResourceResponse.java

```diff
@@ -0,0 +1,133 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.service.resource.model;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+import java.util.Objects;
+import java.util.Optional;
+
+/**
+ * This class is a container for {@code ResourceResponse} and {@code AuditErrorMessage} during stream processing.
+ * Under normal conditions only one of these will be non-null, indicating failed or successful processing.
+ */
+public final class AuditableResourceResponse {
+
+    @JsonProperty("resourceResponse")
+    private final ResourceResponse resourceResponse;
+    @JsonProperty("auditErrorMessage")
+    private final AuditErrorMessage auditErrorMessage;
+
+    @JsonCreator
+    private AuditableResourceResponse(
+            final @JsonProperty("resourceResponse") ResourceResponse resourceResponse,
+            final @JsonProperty("auditErrorMessage") AuditErrorMessage auditErrorMessage) {
+        this.resourceResponse = resourceResponse;
+        this.auditErrorMessage = auditErrorMessage;
+    }
+
+    /**
+     * Chain any errors from previous stream elements
+     *
+     * @param audit the previous audit or null
+     * @return a new instance of this object
+     */
+    public AuditableResourceResponse chain(final AuditErrorMessage audit) {
+        return Optional.ofNullable(audit).map(message -> Builder.create()
+                .withAuditErrorMessage(message))
+                .orElse(this);
+    }
+
+    @Override
+    public boolean equals(final Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        AuditableResourceResponse that = (AuditableResourceResponse) o;
+        return Objects.equals(resourceResponse, that.resourceResponse) &&
+                Objects.equals(auditErrorMessage, that.auditErrorMessage);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(resourceResponse, auditErrorMessage);
+    }
+
+    public ResourceResponse getResourceResponse() {
+        return resourceResponse;
+    }
+
+    public AuditErrorMessage getAuditErrorMessage() {
+        return auditErrorMessage;
+    }
+
+    /**
+     * The static builder
+     */
+    public static class Builder {
+
+        /**
+         * The creator function
+         *
+         * @return the composed immutable object
+         */
+        public static IUserResponse create() {
+            return AuditableResourceResponse::new;
+        }
+
+        /**
+         * Compose with {@code UserResponse}
+         */
+        public interface IUserResponse {
+            /**
+             * Compose value
+             *
+             * @param audit value or null
+             * @return value object
+             */
```

Comment:
javadoc is a bit lacking in these interfaces

Reply:
Updated